src/compiler/compile/render_dom/wrappers/InlineComponent/index.ts

```diff
@@ -579,4 +540,85 @@ export default class InlineComponentWrapper extends Wrapper {
 			);
 		}
 	}
+
+	private create_mount_chunk(
+		block: Block,
+		parent_node: Identifier,
+		name: Identifier,
+		css_custom_properties_wrapper: Identifier | null
+	) {
+			if (css_custom_properties_wrapper) {
+				if (parent_node) {
+					block.chunks.mount.push(b`@append(${parent_node}, ${css_custom_properties_wrapper})`);
+					if (is_head(parent_node)) {
+						block.chunks.destroy.push(b`@detach(${css_custom_properties_wrapper});`);
+					}
+				} else {
+					block.chunks.mount.push(b`@insert(#target, ${css_custom_properties_wrapper}, #anchor);`);
+					// TODO we eventually need to consider what happens to elements
+					// that belong to the same outgroup as an outroing element...
+					block.chunks.destroy.push(b`if (detaching && ${name}) @detach(${css_custom_properties_wrapper});`);
+				}
+			}
+
+			const target = css_custom_properties_wrapper || parent_node || '#target';
+			const anchor = css_custom_properties_wrapper ? 'null' : (parent_node ? 'null' : '#anchor');
+			if (this.node.name === 'svelte:component') {
+				block.chunks.mount.push(b`if (${name}) @mount_component(${name}, ${target}, ${anchor});`);
+			} else {
+				block.chunks.mount.push(b`@mount_component(${name}, ${target}, ${anchor});`);
+			}
```

Comment:
i see that before calling the `create_mount_chunk` we already did check for `this.node.name === 'svelte:component'`, now we are checking this again, feels a bit messy.

also there's quite a lot of level of nesting in this function, handling multiple scenarios, im thinking of maybe a different way of grouping functions and refactoring them?

i took the liberty to actually try organise it myself, see https://github.com/baseballyama/svelte/pull/4, what do you think about it?

Reply:
Sorry for the late reply.
Yeah, complexity is less in your PR. So I merged.
Now I will update the PR according to your comments.
Thank you!